master/src/main/scala/org/apache/celeborn/service/deploy/master/Master.scala

```diff
@@ -624,12 +627,48 @@ private[celeborn] class Master(
     context.reply(OneWayMessageResponse)
   }
 
+  private def computeUserResourceConsumption(userIdentifier: UserIdentifier)
+      : ResourceConsumption = {
+    val current = System.currentTimeMillis()
+    if (userResourceConsumption.containsKey(userIdentifier)) {
+      val resourceConsumptionAndUpdateTime = userResourceConsumption.get(userIdentifier)
+      if (current - resourceConsumptionAndUpdateTime._2 > conf.computeUserResourceResourceConsumptionInterval) {
+        val newResourceConsumption = statusSystem.workers.asScala.flatMap { workerInfo =>
+          workerInfo.userResourceConsumption.asScala.get(userIdentifier)
+        }.foldRight(ResourceConsumption(0, 0, 0, 0))(_ add _)
+        userResourceConsumption.put(userIdentifier, (newResourceConsumption, current))._1
+      } else {
+        resourceConsumptionAndUpdateTime._1
+      }
+    } else {
+      val newResourceConsumption = statusSystem.workers.asScala.flatMap { workerInfo =>
+        workerInfo.userResourceConsumption.asScala.get(userIdentifier)
+      }.foldRight(ResourceConsumption(0, 0, 0, 0))(_ add _)
+      userResourceConsumption.put(userIdentifier, (newResourceConsumption, current))._1
+    }
+  }
+
   private def handleCheckQuota(
       userIdentifier: UserIdentifier,
       context: RpcCallContext): Unit = {
-    val userResourceConsumption = statusSystem.workers.asScala.flatMap { workerInfo =>
-      workerInfo.userResourceConsumption.asScala.get(userIdentifier)
-    }.foldRight(ResourceConsumption(0, 0, 0, 0))(_ add _)
+    val userResourceConsumption = computeUserResourceConsumption(userIdentifier)
```

Comment:
Will this value be updated when each time PrometheusServlet try to get the newest gauge? I did 2 tests pasted below on Scala REPL, but seems the value won't change when each time call it.
```
var a = collection.mutable.ArrayBuffer[Int]()
val b = a.length
print(b)
a.append(1)
print(b)
```
```
var a = collection.mutable.ArrayBuffer[Int]()
def c = a.length
val b = c
print(b)
print(c)
a.append(1)
print(b)
print(c)
```

Reply:
Done